File: src/main/java/com/asynchronousmontser/monstermusic/model/Song.java
package com.asynchronousmontser.monstermusic.model;

import javax.persistence.*;

@Entity
@Table(name = "song")
public class Song {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id")
    private User creator;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "singer_id")
    private Singer singer;

    public Song() {
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public User getCreator() {
        return creator;
    }

    public void setCreator(User creator) {
        this.creator = creator;
    }

    public Singer getSinger() {
        return singer;
    }

    public void setSinger(Singer singer) {
        this.singer = singer;
    }
}

File: client/src/model/song.ts
import {Singer} from "./singer";
import {User} from "./user";

export class Song {
  id: number;
  creator: User;
  createdDate: string;
  self: string;
  link: string;

  name: string;
  singer: Singer;

}

export class DtoSong {
  name: string;
  singer: string;
  link: string;
}
